### DISCUSSION TITLE
Anonimizar movimentações bancárias da APyB para torna-las públicas

### DISCUSSION THREAD

**Discussion by muriloviana** [apyb/comunidade#14] [Geral]
Olá comunidade o/

Uma das primeiras tarefas da nova diretoria da APyB é disponibilizar toda a movimentação bancária da associação de forma pública, como vem sido discutido nesta issue https://github.com/apyb/tarefas/issues/1. Resumindo, a ideia inicial é criarmos um repositório com os extratos mensais da associação, em arquivos CSV, para que qualquer pessoa da comunidade tenha acesso a estes dados e possa usufruir dos mesmos da maneira que melhor entender. Como por exemplo, criando um portal de transparência, uma api e etc.

Tendo em vista a LGPD e o máximo de transparência possível para a comunidade, surgiu um questionamento na última reunião da diretoria: **o quão anonimizado devem ser os dados públicos das movimentações bancárias da APyB?**

Acredito que os valores, datas e horas destas movimentações possam ser públicas sem problema, mas não tenho certeza do quão sensível são os dados pessoais. Por exemplo, um reembolso feito para uma pessoa durante a organização de determinado evento, deveria constar nos dados públicos da movimentação o nome completo e o CPF da pessoa que recebeu o reembolso? O pagamento de um determinado serviço, deveria constar a razão social e o CNPJ da empresa terceira?

Gostaria de propor uma data limite (20d a partir da publicação) para chegarmos em um acordo comum sobre este tema, de forma a não prolonga-lo infinitamente 😅

**Comment by gabubellon** [comment]
@muriloviana Tudo bem ?
Primeira coisa, nos temos hoje ou conseguimos ter fácil (posso ajudar nisso) um dicionário de dados do que vem nesse extrato ?
Ex:
Nome-CPF/CNPJ-TIPO-VALOR-DATA

Falando um pouco da experiência que tenho desses dados, o penso que:
*  Reembolso para as pessoas fisicas devem ser mascarados os CPFs (o nome podemos deixar uma vez que as pessoas que recebem o reembolso são membros da ApyB e essa info é aberta)
* CNPJs de empresas e nomes devem ser mantidos abertos, uma vez que é um contrato e não vejo porque esses dados serem mascardos, podem ser usados até como base para outras pessoas/eventos em saber o custo da empresa
* Pagamentos de imposto/tributos e afins devem ser extremamente abertos, para mostrar que existem e como impactam
* Boletos e afins , se forem de reembolsos podem ser mascarados, se forem de clientes podemos abrir

Em um dos meus empregos trabalhei em um modelo de categorização desses tipos de lançamentos, podemos pensar em criar uma "anomalizador"  especifico da Apyb

**Reply by gabubellon** [reply]
Apenas complementando **VALORESS** seram sempre abertos, talvez a origem/destino precise ser anomizado

**Reply by alynnefs** [reply]
seria interessante ter também a descrição do valor, por exemplo "compra do banco de imagens", "patrocínio ouro", etc

**Reply by gabubellon** [reply]
@alynnefs essa seria a ideia de criar uma solução simples de categorização
@muriloviana sabe se atualmente no BB da pra categorizar e customizar essas categorias ? E se sim , se elas são exportáveis no extrado csv ?

**Reply by rougeth** [reply]
> uma vez que as pessoas que recebem o reembolso são membros da ApyB

Não é uma regra, podemos ter reembolsos para pessoas não associadas. Nem todo mundo da organização faz parte da associação, por exemplo.

> seria interessante ter também a descrição do valor, por exemplo "compra do banco de imagens", "patrocínio ouro", etc

Eu acho que estamos ainda um passo atrás de qualquer manipulação desses dados, que é simplesmente disponibilizá-los. @muriloviana talvez especificar quais são os campos disponíveis e alguns exemplos ajude a deixar mais claro.

**Reply by gabubellon** [reply]
> Não é uma regra, podemos ter reembolsos para pessoas não associadas. Nem todo mundo da organização faz parte da associação, por exemplo.
Bom nesse caso eu acho então tem a gente tem que anonimizar mesmo os CPFS, CNPJs e publico sempre.


E entendo que esse seja o passo inicial mesmo , então tem um sample/colunas pode ajudar a dizer o que precisa ficar aberto ou não
@muriloviana se precisar pegar algo de reembolso pode pegar dos meus para exemplo (só não deixa meu CPF :P)

**Comment by muriloviana** [comment]
Conforme solicitado, para auxiliar na discussão segue o formato e um exemplo de CSV.

O formato do CSV:

```
Agencia;ContaCorrent;Vazio;DataBalancete;DataMovimento;AgenciaOrigem;Lote;Documento;CodigoTipo;Tipo;ValorEmCentavos;CreditoOuDebito;Descricao
```

CSV de extrato fictício do mês 2021-11:

```csv
00892;000000781266; ;27102021;27102021;0000;00000;00000000000000000;000;Saldo Anterior;00000000012684647;C;
00892;000000781266; ;05112021;05112021;0000;14397;00000000627568009;900;PIX - Recebido;00000000000028335;C;05/11 13:44 00012345678900 Murilo Viana
00892;000000781266; ;05112021;05112021;0000;13105;00000000000110501;144;PIX - Enviado;00000000000051717;D;05/11 13:15 Murilo Viana
00892;000000781266; ;12112021;12112021;0000;13105;00000000000111201;375;Impostos;00000000000360000;D;PREF MUN CAXIAS DO SUL
00892;000000781266; ;17112021;17112021;0000;14175;00000000179283465;976;TED-Crédito em Conta;00000000001000000;C;123 1234 12345678000199 SHOSTNERS AND SHOSTNERS LTDA
00892;000000781266; ;30112021;30112021;0000;00000;00000000000000000;999;S A L D O;00000000013301265;C;
```

**Comment by rennerocha** [comment]
Não podemos esquecer que não é apenas a conta bancária do Banco do Brasil que precisa ter seus extratos tornados públicos. A APyB também possui uma conta no PagSeguro onde dinheiro é mantido e que precisamos tornar públicos os extratos.

**Reply by muriloviana** [reply]
Exatamente @rennerocha. Apenas complementando a informação, precisamos tornar públicos os extratos referentes à:
* Conta corrente no Banco do Brasil
* Investimento financeiro no Banco do Brasil
* Conta no PagSeguro

**Comment by rennerocha** [comment]
@muriloviana 

> Tendo em vista a LGPD e o máximo de transparência possível para a comunidade, surgiu um 
> questionamento na última reunião da diretoria: o quão anonimizado devem ser os dados públicos 
> das movimentações bancárias da APyB?

Eu acho que essa decisão (o que deve ser anonimizado ou não) para os associados, não deveria ser definido em uma discussão aqui. Na minha opinião, a diretoria deveria entrar em contato com algum advogado (com alguma experiência em LGPD de preferência) e obter um parecer jurídico adequado. Assim temos uma segurança maior do que pode ser divulgado para os Associados, o que pode ser divulgado para o público em geral e o que deveríamos tornar anônimo.

**Pagar** uma consulta de um advogado e obter uma informação correta, de um profissional competente é na minha opinião a única maneira de fazermos tudo certo de acordo com a legislação evitando assim futuros problemas jurídicos para a Associação e sua diretoria e é um custo que conseguimos pagar sem problemas, tendo em vista o caixa que temos.

**Comment by cuducos** [comment]
> Pagar uma consulta de um advogado e obter uma informação correta, de um profissional competente é na minha opinião a única maneira de fazermos tudo certo de acordo com a legislação

Concordo e adiciono:

Pagar uma consulta de um advogado e obter uma informação correta, de um profissional competente é na minha opinião a única maneira de fazermos tudo da maneira mais **objetiva** possível

**Comment by muriloviana** [comment]
A grande vantagem de trazer essas discussões para a comunidade, ao invés de te-las fechadas apenas entre a diretoria e conselhos, é de trazer diferentes pontos de vista e assim melhorar cada vez mais o trabalho da associação. Dito isso, a sugestão levantado por @rennerocha e apoiada por @cuducos é muito boa e se quer havia passado pela minha mente rs!

Mantendo a ideia de transparência, irei atrás de, ao menos três, orçamentos de profissionais de direito especializados em LGPD, e trago para essa mesma discussão e assim decidimos qual a melhor opção para realizar tal serviço.

P.S. Sugestões de profissionais da área são super bem-vindas 😉

**Reply by gabubellon** [reply]
@muriloviana vou indicar o https://blconsultoriadigital.com.br/, no caso a https://www.instagram.com/grazielabrandao.adv/ que uma das fundadoras

**Reply by rennerocha** [reply]
@gabubellon fui conversar com um amigo aqui de Campinas que a esposa é especialista em LGPD para pedir o contato dela e descobri que é a Graziela mesmo. Podem agendar uma consulta com eles que também indico!

**Reply by gabubellon** [reply]
@rennerocha big small world, conheço a  Graziela da época de Campinas também , quando ela trabalhava em uma empresa de cripto

**Reply by cuducos** [reply]
@jedibruno, tu se interessaria em prestar essa consultoria?

**Reply by jedibruno** [reply]
Bom dia, pessoal! 

@cuducos , tenho interesse, sim. Será um prazer conversar a respeito!

**Reply by tiidadavena** [reply]
Posso indicar o Arquimedes Rocha, ele atua como consultor e também aceita conversar sobre, caso tenham interesse. Ele não tem GIthub, mas o email dele é arquimedes@academy.cin.ufpe.br

**Reply by muriloviana** [reply]
@gabubellon @rennerocha @cuducos @jedibruno @tiidadavena enviei email/mensagem para as indicacões que surgiram aqui, muuuito obrigado! Agora é aguardar o retorno com os orçamentos.

P.S. se também quiserem dar um toque nessas pessoas, é super bem-vindo \o/

**Comment by muriloviana** [comment]
Passando para agradecer a todos que participaram da discussão e informar que: a diretoria da APyB, analisou os orçamentos dos profissionais sugeridos nesta discussão e, escolheu o Bruno Morassutti para nos auxiliar nesta tarefa de anonimização. Portante, nesta quarta-feira (13/04/2022) daremos início ao processo de disponibilizar, de forma pública, as movimentações bancárias da associação.
